### ISSUE TITLE
File preview not clickable with Fronteend Edit

### ISSUE THREAD

**Issue by upstreamplugin** [upstreamplugin/UpStream#181]
When a file is added to a Bug, on the frontend by default (without Upstream Frontend Edit), the file preview is clickable. 

However, using Upstream Frontend Edit, if I edit a Bug, the file preview on the modal is not clickable.

Since the dropdown button is very small, users tend to click Edit. They may be a bit disappointed to find that the file preview is small and not clickable.

Steps to reproduce ...

- Scroll all the way down to the Files section
- Click the dropdown button on Test
- Click the file preview. It will open on a new window
- Go back to Files > Test
- Click Edit on Test
- The file preview on the modal window is not clickable

![screen shot 2017-09-25 at 7 33 11 am](https://user-images.githubusercontent.com/26938537/30806454-cc8ca49a-a1c3-11e7-8a2e-182f56f696d7.png)

On upstream-frontend-edit.js:

Find this
`var file = $( this ).find( '.itemfile' ).clone(); `

Directly below that line, add this code:

`var fileParent = $( this ).find( '.itemfile' ).parent();`

Find and replace this line:
`$form.find( '.file-preview div' ).append( existing ).append( file ); `

Replace it with this code:

```
if( fileParent.prop( 'tagName' ) == 'A' ) {
$form.find( '.file-preview div' ).append( existing ).append( fileParent );
}
else {
$form.find( '.file-preview div' ).append( existing ).append( file );
}

```

**Comment by deenison** [comment]
Thanks, although the suggested code changes breaks the file preview on the bugs list.

A more appropriate solution would be shipped on the `v1.3.8` of [Frontend Edit](https://upstreamplugin.com/extensions/frontend-edit) extension.

**Comment by awijasa** [comment]
Hello @deenison,

I looked into this again, and modified my workaround. Posting it here just in case it can be any help:

On upstream-frontend-edit.js:

Find this
`var file = $( this ).find( '.itemfile' ).clone();`

Directly below that line, add this code:

`var fileParent = $( this ).find( '.itemfile' ).parent().clone();`

I forgot to use .clone() before. Without clone(), editing a bug or file on the frontend will remove the original file preview.

Find and replace these lines:
```
$form.find( '.file-preview div' ).append( existing ).append( file );
$form.find( '.file-preview' ).show();
```
I found that directly show()ing the file-preview will cause a white flicker at the bottom left of my screen. So, there needs to be a timeout before showing the file-preview (around 700ms).

Replace it with this code:

```
if( fileParent.prop( 'tagName' ) == 'A' ) {
    $form.find( '.file-preview div' ).append( existing ).append( fileParent );
}
else {
    $form.find( '.file-preview div' ).append( existing ).append( file );
}

setTimeout( function() { $form.find( '.file-preview' ).slideDown() }, 700 );
```

Finally, before `$form.find('.file-preview div').html('');`, add `$form.find( '.file-preview' ).hide();`. This will hide the file-preview so that it won't cause the white flicker.

Please let me know if you have any question.

Thanks,

@awijasa
